sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessageSerializer.java

```diff
@@ -225,6 +234,62 @@ public int getSize(Message amqpMessage) {
         return amqpMessage;
     }
 
+    private Map<String, Object> convertToDescribedType(Map<String, Object> propertiesValue) {
+        for (Map.Entry<String, Object> entry : propertiesValue.entrySet()) {
+            Object value = entry.getValue();
+            if (value instanceof URI) {
+                propertiesValue.put(entry.getKey(), new ServiceBusDescribedType(ServiceBusConstants.URI_SYMBOL, ((URI) value).toString()));
+            } else if (value instanceof OffsetDateTime) {
+                propertiesValue.put(entry.getKey(), new ServiceBusDescribedType(ServiceBusConstants.OFFSETDATETIME_SYMBOL, convertToTickTime((OffsetDateTime) value)));
+            } else if (value instanceof Duration) {
+                propertiesValue.put(entry.getKey(), new ServiceBusDescribedType(ServiceBusConstants.DURATION_SYMBOL, convertToTickTime((Duration) value)));
+            }
+        }
+        return propertiesValue;
+    }
+
+    private Map<String, Object> convertToOriginType(Map<String, Object> propertiesValue) {
+        for (Map.Entry<String, Object> entry : propertiesValue.entrySet()) {
+            Object value = entry.getValue();
+            if (value instanceof DescribedType) {
+                DescribedType describedType = (DescribedType) value;
+                if (ServiceBusConstants.URI_SYMBOL.equals(describedType.getDescriptor())) {
+                    try {
+                        URI uri = URI.create((String) describedType.getDescribed());
+                        propertiesValue.put(entry.getKey(), uri);
+                    } catch (Throwable th) {
+                        propertiesValue.put(entry.getKey(), (String) describedType.getDescribed());
+                    }
+                } else if (ServiceBusConstants.OFFSETDATETIME_SYMBOL.equals(describedType.getDescriptor())) {
+                    propertiesValue.put(entry.getKey(), convertToOffsetDateTime(describedType.getDescribed()));
+                } else if (ServiceBusConstants.DURATION_SYMBOL.equals(describedType.getDescriptor())) {
+                    propertiesValue.put(entry.getKey(), convertToDuration(describedType.getDescribed()));
+                }
+            }
+        }
+        return propertiesValue;
+    }
+
+    public static Long convertToTickTime(OffsetDateTime offsetDateTime) {
```

Comment:
Use private for convert method?
Or move to a util class as public static method?

Reply:
I'll change it to private method.